content/tutorials/guides/how-to-get-pim-category-tree.md

```diff
@@ -0,0 +1,325 @@
+        </div>
+    </div>
+</div>
+
+
+## Context
+
+A category tree is typically a graph where nodes are categories with or without sub-categories.
+The first node - the root of the graph - is the root category of the category tree.
+
+![relationship schema](/img/getting-started/synchronize-pim-products/step-3-objects-relationship-schema.svg)
+
+:::tips
+Get the big picture <a href="/getting-started/synchronize-pim-products-6x/step-0.html" target="_blank" rel="noopener noreferrer">here</a>.
+:::
+Fortunately, getting the whole category is as simple as querying the correct API endpoint.
+
+## Fetch the category tree
+
+### 0. Initialization
+
+```php [activate:PHP]
+
+function buildApiClient(): GuzzleHttp\Client
+{
+    $pimUrl = 'https://url-of-your-pim.com';
+    $appToken = 'your_app_token'; // Token provided during oAuth steps
+
+    // If you haven't done it yet,
+    // please follow the Guzzle official documentation to install the client 
+    // https://docs.guzzlephp.org/en/stable/overview.html#installation
+
+    // Set your client for querying Akeneo API as follows
+    $client = new \GuzzleHttp\Client([
+        'base_uri' => $pimUrl,
+        'headers' => ['Authorization' => 'Bearer ' . $appToken],
+    ]);
+}
+```
+
+```javascript [activate:NodeJS]
+
+// Install the node-fetch library by following the official documentation:
+// https://www.npmjs.com/package/node-fetch
+import fetch from 'node-fetch';
+
+const pimUrl = 'https://url-of-your-pim.com';
+const accessToken = 'your_app_token'; // Token provided during oAuth steps
+
+// Set your client for querying Akeneo API as follows
+async function get(url, accessToken) {
+  return await fetch(url, {
+    headers: {
+      'Authorization': `Bearer ${accessToken}`
+    }
+  });
+}
+```
+
+### 1. Get the PIM structure by fetching a channel from API
+
+Workflow:
+
+![synchronisation steps](/img/getting-started/synchronize-pim-products/step-3-steps-schema.svg)
+
+Collect flat category tree from PIM API:
+
+```php [activate:PHP]
+
+function getCategories(): void
+{
+    $client = buildApiClient();
+
+    $rootCategoryCode = 'master';
+    $maxItems = 100;
+
+    $nextUrl = sprintf(
+        '/api/rest/v1/categories'
+        . '?with_position=true'
+        . '&search={"parent":[{"operator":"=","value":"%s"}]}'
+        . '&limit=%s',
+        $rootCategoryCode,
+        $maxItems
+    );
+
+    $categoryPages = [];
+
+    do {
+        $response = $client->get($nextUrl);
+        $data = json_decode($response->getBody()->getContents(), true);
+        $categoryPages[] = $data['_embedded']['items'];
+
+        $nextUrl = $data['_links']['next']['href'] ?? null;
+    } while (
+        $nextUrl
+    );
+
+		// Adds root category to complete the list.
+    $categoryPages[][] = getRootCategory($rootCategoryCode);
+
+    $categories = array_merge(...$categoryPages);
+
+    saveCategories($categories);
```

Comment:
add comments: the user doesn't know what is this function

Reply:
php-guided-tutorial: done